central/reports/service/v2/service_impl.go

```diff
@@ -115,3 +122,60 @@ func (s *serviceImpl) GetReportHistory(ctx context.Context, req *apiV2.GetReport
 	}
 	return &res, nil
 }
+
+func (s *serviceImpl) RunReport(ctx context.Context, req *apiV2.RunReportRequest) (*apiV2.RunReportResponse, error) {
+	if req.GetReportConfigId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Report configuration ID is empty")
+	}
+	slimUser := authn.UserFromContext(ctx)
+	if slimUser == nil {
+		return nil, errors.New("Could not determine user identity from provided context")
+	}
+	config, found, err := s.reportConfigStore.GetReportConfiguration(ctx, req.GetReportConfigId())
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error finding report configuration %s", req.GetReportConfigId())
+	}
+	if !found {
+		return nil, errors.Wrapf(errox.NotFound, "Report configuration id not found %s", req.GetReportConfigId())
+	}
```

Comment:
The implementation of `RunReport` and `CancelReport` is pretty inconsistent. One is doing the validation and the other is not.

Reply:
```
	config, found, err := s.reportConfigStore.GetReportConfiguration(ctx, req.GetReportConfigId())
	if err != nil {
		return nil, errors.Wrapf(err, "Error finding report configuration %s", req.GetReportConfigId())
	}
	if !found {
		return nil, errors.Wrapf(errox.NotFound, "Report configuration id not found %s", req.GetReportConfigId())
	}
```

There is no guarantee that the report configuration that is checked above still exists in the database when the job is handed over to the scheduler.

Reply:
Good catch, that is correct. Checking whether the config still exists at every step of report generation would not be ideal too. I think it would be better to prevent deletion of a report config for which there is already a WAITING OR PREPARING report.  I can make that change to the report config service.

Reply:
Moved `CancelReport` validations out of scheduler and to the service layer.